frappe/public/js/frappe/model/perm.js

```diff
@@ -34,9 +34,7 @@ $.extend(frappe.perm, {
 
 	doctype_perm: {},
 
-	has_perm: (doctype, permlevel, ptype, doc) => {
-		if (!permlevel) permlevel = 0;
-
+	has_perm: (doctype, permlevel = 0, ptype = "read", doc) => {
```

Comment:
is this allowed? having a param with no defaults after params with defaults :thinking: 

and even if it's allowed we should probably not do this?

Reply:
> and even if it's allowed we should probably not do this?

what's the concern? We are checking for minimum possible permission for nothing is specified.

Reply:
It's only only a problem in python not JS.